book_source/adv_user_guide_web/Benchmarking.Rmd

```diff
@@ -0,0 +1,84 @@
+# Benchmarking
+
+Benchmarking is the process of comparing model outputs against either experimental data or against other model outputs as a way to validate model performance. 
+We have a suit of statistical comparisons that provide benchmarking scores as well as visual comparisons that help in diagnosing data-model and/or model-model differences.
+
+There are currently two ways to use the benchmarking tools. 
+
+1. **New Run**: Set up a new model run through pecan interface. After performing the run, immediately load up input data that can be compared to the model outputs. This can be done using the online shiny benchmarking app.
+
+2. **Clone Run**: Set up a new model run in pecan that will use preexisting run settings that have been stored in BETY as  a "reference run" record. This setup is more involved and currently has to be done by hand by editing the pecan.xml document in Rstudio. 
+
+## Before benchmarking
+
+All the data that you want to compare with model runs must be registered in the database. This is currently a step that must be done by hand either from the command line or through the online BETY interface. 
+I.e. the data must have three records:
+
+1. An input record (Instructions [here](#NewInput))
+
+2. A database file record (Instructions [here](#NewInput)))
+
+3. A format record (Instructions [here](#NewFormat)))
+
+
+## Benchmarking in pecan.xml
+
+The `pecan.xml` has an _optional_ benchmarking section. Ultimately, the user should not have to edit this document by hand, but it is still necessary for cloning runs. 
+
+Below are all the possible benchmarking sections. 
+If you provide the ensemble id, then you do not need to provide the reference run id and vice versa.
+
+`<benchmarking>` settings:
+
+- `ensemble_id`: the id of the ensemble that you will be using - the settings from this ensemble will be saved in a reference run record and then `ensemble_id` will be replaced with `reference_run_id` 
+- `input_id`: the id(s) of the benchmarking data  (required)
+- `variable_id`: the id(s) of the variable(s) of interest within the data. If you leave this blank, all variables that are shared between the input and model output will be used. 
+- `metric_id`: the id(s) of the metric(s) to be calculated. If you leave this blank, all metrics will be used. 
+- `new_run`: TRUE = create new run, FALSE = use existing run (required)
+
+Example:
+
+```xml
+ <benchmark>
+     <ensemble_id>1000004796</ensemble_id> 
+     <input_id>1000008121</input_id>
+     <variables>
+       <variable_id>411</variable_id> 
+     </variables>
+     <metrics>
+       <metric_id>1000000003</metric_id> 
+     </metrics>
+     <new_run>TRUE</new_run>
+   <reference_run_id>1000000050</reference_run_id>
```

Comment:
This section refers to the tag as ```<benchmark>``` while the section above says it's ```<benchmarking>```. One of these is wrong. I'd prefer the former.

This section also uses the ```<reference_run_id>``` tag without documenting it.

The text above refers to various things as 'required' that I don't think are (or at least I don't think should be) and continues to confuse the distinction between run DEFINITION and run EXECUTION.  I'd recommend having an very explicit separation of these two uses  in the documentation since it has become a major source of confusion for both users and developers and I feel the current documentation will just reinforce that confusion. Since the primary use of benchmarking is for the EXECUTION of existing benchmarks, and because few users will create their benchmarks via XML, I think that should go first. In which case, as far as I understand, we just need ```<reference_run_id>``` documented and to clarify whether a workflow that runs multiple reference runs should have ```<benchmark>``` sections (each with only one ```<reference_run_id>```, or just one section that has multiple ```<reference_run_id>```'s.  Personally, I think the latter is simpler, but if that's not how the code is set up now, then you should document the way the code is set up now.